### PR TITLE
feat: prerelease type

### LINKED ISSUE #2082: Prerelease type for new prerelease versions
The prerelease version updater is a great feature that allows to bump the prerelease type in a particular way. However, when the version doesn't have a prerelease part, it doesn't do anything. Therefore, it would be nice to have a way that allows to have bumps such as 
- `1.3.3` to `1.3.4-PTPT` fore patches
- `1.3.3` to `1.4.0-PTPT` for features
- and `1.3.3` to `2.0.0-PTPT` for breaking changes

where `PTPT` is a custom prerelease type.

### PR DESCRIPTION
**Pull request by kalosisz**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/release-please/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #2082 🦕 by introducing an argument to the `prerelease` versioning strategy.

It also deals with bumping versions with prerelease, but not having a number in the prerelease part in the following ways  
- `2.0.0-beta` will get bumped to `2.0.0-beta.1` if there are breaking changes or less
- `1.2.0-beta` will get bumped to `1.2.0-beta.1` if there are featurer or less
- `1.2.2-beta` will get bumped to `1.2.2-beta.1` if there are only patches.

The current behaviour of prelrease, ie. bumping `1.1.1-beta01` to `1.1.1-beta02` is preserved.

### CODE REVIEW

File: docs/manifest-releaser.md [hunk 1]
```diff
@@ -172,6 +172,9 @@ defaults (those are documented in comments)
   // absence defaults to false
   "bump-patch-for-minor-pre-major": true,
 
+  // setting the type of prerelease in case of prerelease strategy
+  "prerelease-type": true,
```
Comment (chingor13) [inline docs/manifest-releaser.md:176]: Should this be a string?

File: schemas/config.json [hunk 2]
```diff
@@ -20,6 +20,10 @@
           "description": "Feature changes only bump semver patch if version < 1.0.0",
           "type": "boolean"
         },
+        "prerelease-type": {
+          "description": "Configuration option for the prerelese versioning strategy. If prerelease strategy used and type set, will set the prerelese part of the version to the provided value in case prerelease part is not present.",
+          "type": "boolean"
```
Comment (chingor13) [inline schemas/config.json:25]: Same here - should this be a string?

File: test/factory.ts [hunk 3]
```diff
@@ -60,19 +61,24 @@ describe('factory', () => {
       expect(await strategy.getComponent()).not.ok;
       expect(strategy.changelogNotes).instanceof(DefaultChangelogNotes);
     });
-    it('should build a with configuration', async () => {
+    it('should build with prerelease type', async () => {
```
Comment (chingor13) [inline test/factory.ts:64]: nit: can we add this as a separate test and leave the existing one?

**Review by chingor13 — CHANGES_REQUESTED** [review]
Thanks for this. A few minor comments/nits.

File: docs/manifest-releaser.md (hunk 1 above)
Comment (kalosisz) [reply docs/manifest-releaser.md]: done

**Comment by kalosisz** [comment]
hey @chingor13, please check the changes :) thanks

**Review by chingor13 — APPROVED** [review]
Thanks!

**Pull request merged**